ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/messagelistpage/helpers/OffboardingHelperTest.java

```diff
@@ -0,0 +1,335 @@
+    }
+
+    @Test
+    public void
+    givenGetOffboardingListItemsWhenCurrentFeedSubmittedViaUINullThenGetOffboardingItemsToSelectRating() {
+        // Arrange
+        Rating.SCORE currentRatingSubmittedViaUI = Rating.SCORE.GOOD;
+        setInternalState(underTest, "currentRatingSubmittedViaUI", currentRatingSubmittedViaUI);
+        String currentFeedbackSubmittedViaUI = null;
+        setInternalState(underTest, "currentFeedbackSubmittedViaUI", currentFeedbackSubmittedViaUI);
+        when(Kayako.getApplicationContext()).thenReturn(context);
+        when(context.getResources()).thenReturn(resources);
+        when(resources.getString(R.string
+                .ko__messenger_input_feedback_rating_instruction_message)).thenReturn("message");
+        // Act
+        List<BaseListItem> result = underTest.getOffboardingListItems("name", true,
+                offboardingHelperViewCallback);
+        // Assert
+        assertFalse(result.isEmpty());
+    }
+
+    @Test
+    public void
+    givenGetOffboardingListItemsWhenFeedbackAndRatingNotNullThenGetOffboardingItemsToSelectRating
+            () {
+        // Arrange
+        Rating.SCORE currentRatingSubmittedViaUI = Rating.SCORE.GOOD;
+        setInternalState(underTest, "currentRatingSubmittedViaUI", currentRatingSubmittedViaUI);
+        String currentFeedbackSubmittedViaUI = "good";
+        setInternalState(underTest, "currentFeedbackSubmittedViaUI", currentFeedbackSubmittedViaUI);
+        // Act
+        List<BaseListItem> result = underTest.getOffboardingListItems("name", true,
+                offboardingHelperViewCallback);
+        // Assert
+        assertTrue(result.isEmpty());
+    }
+
+    @Test
+    public void givenResetRatingAddedByUserThenIdOfLastRatingSetToZero() throws Exception {
+        // Arrange
+        AtomicLong mIdOfLastRatingAdded = mock(AtomicLong.class);
+        setInternalState(underTest, "mIdOfLastRatingAdded", mIdOfLastRatingAdded);
+        // Act
+        invokeMethod(underTest, "resetRatingAddedByUser");
+        // Assert
+        assertTrue(0L == mIdOfLastRatingAdded.get());
+    }
+
+    @Test
+    public void givenConvertWhenScoreBadThenInputFeedbackRatingBad() throws Exception {
+        // Arrange
+        Rating.SCORE score = Rating.SCORE.BAD;
+        // Act
+        InputFeedback.RATING result = invokeMethod(underTest, "convert", score);
+        // Assert
+        assertEquals(InputFeedback.RATING.BAD, result);
+    }
+
+    @Test
+    public void givenConvertWhenScoreNullThenException() throws Exception {
+        // Arrange
+        Rating.SCORE score = null;
+        // Assert
+        expectedException.expect(Exception.class);
+        // Act
+        invokeMethod(underTest, "convert", score);
+    }
+
+    @Test
+    public void givenConvertFromInputFieldRatingWhenRatingBadThenScoreBad() throws Exception {
+        // Arrange
+        InputFeedback.RATING rating = InputFeedback.RATING.BAD;
```

Comment:
❗️ This code is using the class field `rating` as a local variable. Local variables should not shadow class fields. Shadowing fields with a local variable or with a method parameter is a bad practice that reduces code readability, as it is confusing to know whether the field or the variable is being used. Consider renaming this local variable.  

          
## The solution in this case

```JAVA
InputFeedback.RATING feedbackRating = InputFeedback.RATING.BAD;

```
 

<a id="0fa27630-5329-11e8-8cbd-e396d4991aea" href="none" codereview-uuid-reserved-tag />